driver-core/src/main/com/mongodb/client/model/TimeSeriesOptions.java

```diff
@@ -104,12 +109,102 @@ public TimeSeriesOptions granularity(@Nullable final TimeSeriesGranularity granu
         return this;
     }
 
+    /**
+     * Returns the maximum time span between measurements in a bucket.
+     *
+     * @param timeUnit the time unit.
+     * @return time span between measurements, or {@code null} if not set.
+     * @since 4.10
+     * @mongodb.server.release 6.3
+     * @mongodb.driver.manual core/timeseries-collections/ Time-series collections
+     * @see #bucketMaxSpan(Long, TimeUnit)
+     */
+    @Nullable
+    public Long getBucketMaxSpan(final TimeUnit timeUnit) {
+        if (bucketMaxSpanSeconds == null) {
+            return null;
+        }
+        return timeUnit.convert(bucketMaxSpanSeconds, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Sets the maximum time span between measurements in a bucket.
+     * <p>
+     * The value of {@code bucketMaxSpanSeconds} must be the same as {@code bucketRoundingSeconds}.
```

Comment:
I agree with your viewpoint. However, as mentioned in the [design doc](https://docs.google.com/document/d/1OFq2jyyMLcOkNcnDXaYKXNfdTzDsHufHBAwSp_HDjIo/edit) we have the requirement:

> If a user sets one value, they must set the other. If set directly, we will enforce that the value of bucketRoundingSeconds is equal to the value of bucketMaxSpanSeconds. The latter restriction could be relaxed in the future.

So keeping both options separate allows for flexibility to relax the requirement.

Reply:
Oh, if this is the requirement at the moment, let's make it more explicit in the docs, because it appears very unintuitive to me. Adding something like _"..., which also means that the options must either be both set or both unset"_ may help.